### PR TITLE
Refactoring / CHES API update

### PR DESCRIPTION
**Pull request by parc-jason**
Do a little refactoring before adding in CHES status screens...
Update handling the latest CHES API results...
Refactor utility methods and excel methods into utils files.
Refactor common constants.
Allow caller to download CHES transactions

# Description

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-z208i4-tools.pathfinder.gov.bc.ca/dashboard?id=nr-messaging-service-showcase-pr-73

File: frontend/src/ches/MergeAbout.jsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            UTC. This is what CHES expects.
+          </li>
+          <li>Also, note down the lefthand side of the body is a listing of the variables you can use in the
+            template. These are the CSV headings (altered if needed by MSSC)
+          </li>
+          <li>In Preview, you can navigate through all the messages that will be delivered.</li>
+          <li><strong>Important:</strong> If you alter the JSON, the Excel data table will go away. This is a
+            one way operation (excel to JSON).
+          </li>
+        </ul>      </div>
```
Comment (loneil) [inline frontend/src/ches/MergeAbout.jsx]: Drop div down to next line

File: frontend/src/utils/Constants.jsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// MSSC related
+export const SENDER_EDITOR_ROLE = 'mssc:sender_editor';
+export const DEFAULT_SENDER = 'NR.CommonServiceShowcase@gov.bc.ca';
+
+// CMSG related
+export const CMSG_MEDIA_TYPES = ['text/plain', 'text/html'];
```
Comment (loneil) [inline frontend/src/utils/Constants.jsx]: I'm generally not sold on arrays of constants. The idea of constant naming is to be able to know the value of what a constant is/is going to do at the points in the code it's executed.
So, as a hypothetical example, 
`if(x == CMSG_MEDIA_TYPES[1]) {do something}`
doesn't tell me it's going to do something based on that value being html the way that 
`if(x == CMSG_MEDIA_TYPES.HTML)` 
does.
In my opinion at least! Maybe I'm missing something obvious. Don't think it requires a change, just a note.

Comment (parc-jason) [reply frontend/src/utils/Constants.jsx]: i'll do it in the next PR.

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-z208i4-tools.pathfinder.gov.bc.ca/dashboard?id=nr-messaging-service-showcase-pr-73

**Pull request merged**